src/NUnitFramework/framework/Constraints/ThrowsExceptionConstraint.cs

```diff
@@ -98,7 +98,12 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         public override ConstraintResult ApplyTo<TActual>(ActualValueDelegate<TActual> del)
         {
 #if ASYNC
-            if (typeof(TActual) == typeof(Task)) return ApplyTo(new AsyncTestDelegate(() => (Task)(object)del.Invoke()));
+            if (typeof(TActual) == typeof(Task))
+                return ApplyTo(new AsyncTestDelegate(() => (Task)(object)del.Invoke()));
+            
+            var typeWrapper = new TypeWrapper(typeof(TActual));
```

Comment:
I'm unsure if this is the best way to do this, but I see this pattern in other places in the codebase